### PR TITLE
[terra-popup] Changed disabledHeader to true as default

### PR DESCRIPTION
**Pull request by adavijit**
### Summary

**What was changed:**
- Enabled close button for popup
- Modified tests for the close button

**Why it was changed:**
Should be able to focus close button to close the popup

### Testing

This change was tested using:

- [x] WDIO
- [x] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-10398

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Review by saket2403 — APPROVED** [review]

**Comment by sugan2416** [comment]
@adavijit

-  Doc examples of popup are causing page break

image

- popup content area has reduced, adding a vertical scroll. Can you check why this has changed and since when?

image
  
- Since all test examples of popup have set isHeaderDisabled to false, shouldn't doc examples be updated as well?

**Comment by sdadn** [comment]
The targetRef prop was being set as [required](https://github.com/cerner/terra-framework/blob/610b70fd9468e3627ca4f96be7640bbaf3791acf/packages/terra-popup/src/Popup.jsx#L41) in #2158. 

This is something that also needs to be reverted.

File: packages/terra-popup/CHANGELOG.md
Comment (sdadn) [inline packages/terra-popup/CHANGELOG.md:1]: We should add a note to 6.84.0 that it has a breaking change that's corrected in 6.86.0

**Comment by saket2403** [comment]
> The targetRef prop was being set as [required](https://github.com/cerner/terra-framework/blob/610b70fd9468e3627ca4f96be7640bbaf3791acf/packages/terra-popup/src/Popup.jsx#L41) in #2158.
> 
> This is also a breaking change that also needs to be reverted.

It was always a required I had made it optional in the previous release and reverted the change. Closing this PR now

**Pull request closed without merging**